backend/players/views.py

```diff
@@ -61,4 +72,85 @@ def create_player(request):
 #             'avatar_url': player.avatar.url
 #         }, status=200)
 
-#     return JsonResponse({'error': 'Invalid request method'}, status=405)
\ No newline at end of file
+#     return JsonResponse({'error': 'Invalid request method'}, status=405)
+
+
+def oauth_redirect(request):
+    # Get OAuth parameters from environment variables
+    client_id = os.environ.get('OAUTH_CLIENT_ID')
+    redirect_uri = os.environ.get('OAUTH_REDIRECT')
+    state = get_random_string(32)
+    request.session['oauth_state'] = state
+    base_url = 'https://api.intra.42.fr/oauth/authorize'
+    params = {
+        'client_id': client_id,
+        'redirect_uri': redirect_uri,
+        'response_type': 'code',
+        'state': state
+    }
+    auth_url = f"{base_url}?{urlencode(params)}"
+    return HttpResponseRedirect(auth_url)
+
+
+def oauth_callback(request):
+
+    if request.method == 'GET':
+        # state = request.GET.get('state')
+        code = request.GET.get('code')
+        if not code:
+             return JsonResponse({'error': 'No code provided'}, status=400)
+        token_response = exchange_code_for_token(code)
+        if 'error' in token_response:
+            return JsonResponse(token_response, status=400)
+        user_data = fetch_42_user_data(token_response['access_token'])
+        if 'error' in user_data:
+            return JsonResponse(user_data, status=400)
+        try:
+            # Create the user
+            user = User.objects.create_user(username=f'{user_data['login']}')
```

Comment:
As I understand, single quote in single quotes should not be a problem starting from Python 3.12.
Still the line can be simplified to: 
```suggestion
            user = User.objects.create_user(username=user_data['login'])
```